backend/helpers/helpers.go

```diff
@@ -13,24 +17,91 @@ func CheckErr(err error) {
 	}
 }
 
-func CheckFields(s interface{}) bool {
+func CheckEmptyString(s interface{}) bool {
 	vals := reflect.ValueOf(s)
 	fieldNum := vals.NumField()
+
 	for i := 0; i < fieldNum; i++ {
 		curVal := vals.Field(i)
-		isSet := curVal.IsValid() && curVal.String() != ""
-		if !isSet {
-			return false
+
+		if curVal.Type().Kind() == reflect.String {
+			isSet := curVal.IsValid() && !curVal.IsZero()
+			if !isSet {
+				return false
+			}
 		}
 	}
 
 	return true
 }
-
-func DBErrorHandling(error error, w http.ResponseWriter, r *http.Request) {
+func DBErrorHandling(error error, w http.ResponseWriter) {
 	if errors.Is(error, gorm.ErrRecordNotFound) {
 		http.Error(w, "404", http.StatusBadRequest)
 	} else {
 		http.Error(w, "500", http.StatusInternalServerError)
 	}
 }
+
+func GetAllTable[T models.AllTables](w http.ResponseWriter, table []T) {
+	result := config.DB.Find(&table)
+	if result.Error != nil {
+		DBErrorHandling(result.Error, w)
+		return
+	}
+	//go yells at you if you don't error check this, better to be safe
+	errJson := json.NewEncoder(w).Encode(&table)
+	if errJson != nil {
+		http.Error(w, "500", http.StatusInternalServerError)
+		return
+	}
+}
+
+func GetById[T models.AllTables](w http.ResponseWriter, table T, Id string) {
```

Comment:
Is this get by Id or get by primaryKey?

Reply:
it'd be primary key, can rename, in our case i think all our primary keys are some form of Id based on atleast the common structure so I named it as so

Reply:
lets rename it primaryKey i think that's more obvious

Reply:
on another note why is it a string, aren't out ids ints

Reply:
gonna make it pk, just cuz go variable naming. 
It works the practically the same if its a string or an int, I may need to cast and like manually handle the string-> int if I make it an integer (which I'd want to do in this function anyways cuz it'd be the same for every pk so I think in that case I'd probably still want to pass it as a string and then handle it as integer). The db will basically spit out key not found if you pass it a string so that's why I left it as opposed to me manually spitting out "id is not an integer" on error so. Lmk if you care enough.

Reply:
Nvm, see new commit for updates to this